File: Business/Models/TaskModel.cs
﻿using Data.Entities;
using System.ComponentModel.DataAnnotations;

namespace Business.Models
{
    public class TaskModel
    {
        public int Id { get; set; }

        [Required(AllowEmptyStrings = false)]
        [StringLength(150)]
        public string Name { get; set; } = string.Empty;

        public string? Description { get; set; }

        [Required]
        public DateTimeOffset StartDate { get; set; }

        public DateTimeOffset? Deadline { get; set; }

        public DateTimeOffset? EndDate { get; set; }      // should be null on creation

        [Required]
        public TaskStatuses Status { get; set; }

        public TaskPriorities? Priority { get; set; }

        [Required]
        public int ProjectId { get; set; }

        public ICollection<int> MessagesIds { get; set; } = new List<int>();

        public ICollection<int> FilesIds { get; set; } = new List<int>();

        public ICollection<int> ExecutorsIds { get; set; } = new List<int>();
    }
}

File: AngularApp/src/app/shared/task/task.model.ts
import { TaskPriority } from "./priority/task-priority";
import { TaskStatus } from "./status/task-status";
import { TaskExtraReducedModel } from "./task-extra-reduced.model";

export interface TaskModel extends TaskExtraReducedModel {
    description: string | null | undefined;

    startDate: string;

    deadline: string | null | undefined;

    endDate: string | null | undefined;

    status: TaskStatus;

    priority: TaskPriority | null | undefined;

    projectId: number;
}
